### PR TITLE
libutee: _utee_panic: add register info

### PR DESCRIPTION
**Pull request by liushiwei007**
Added the ability to save register information
on the stack in function _utee_panic.

Signed-off-by: liushiwei <liushiwei@eswincomputing.com>

### CODE REVIEW

**Comment by liushiwei007** [comment]
@maroueneboubakri  Please see if this meets your requirements for panic function.

File: lib/libutee/arch/riscv/sub.mk [hunk 1]
```diff
@@ -1,3 +1,4 @@
 cppflags-y += -I$(sub-dir)/../..
+global-incdirs-y += ../../../../core/arch/riscv/include
```
Comment (jenswi-linaro) [inline lib/libutee/arch/riscv/sub.mk:2]: We don't reach into other parts of the source tree like this. If there's something in `core/arch/riscv/include` that's needed in libutils then it should probably be kept in a libutils include dir instead.

Comment (liushiwei007) [reply lib/libutee/arch/riscv/sub.mk:2]: > We don't reach into other parts of the source tree like this. If there's something in `core/arch/riscv/include` that's needed in libutils then it should probably be kept in a libutils include dir instead.

I refer to the macro store_xregs, It is kept in the core/arch/riscv/include/riscv_macros. S, compile ldelf can access to, but when compile user mode library can't access.  Can I store a copy of riscv_macros. S in the library header file?

Comment (jenswi-linaro) [reply lib/libutee/arch/riscv/sub.mk:2]: No copies, please. If riscv.h and riscv_macros.S are needed outside /core then I believe `lib/libutils/ext/arch/riscv/include` should be a good place to keep them.

Comment (liushiwei007) [reply lib/libutee/arch/riscv/sub.mk:2]: I don't see this directory yet, do you mean to create this directory, change some compilation options, and make it accessible to outside and core?

Comment (jenswi-linaro) [reply lib/libutee/arch/riscv/sub.mk:2]: Yes, please.

Comment (liushiwei007) [reply lib/libutee/arch/riscv/sub.mk:2]: Ok, I try it.

**Comment by liushiwei007** [comment]
@maroueneboubakri @jenswi-linaro 
riscv_macros.S and riscv.h are updated in #5714 ，and I also cited the changes in this update，Whether wait  #5714 merged, then move the two files to the lib/libutils/ext/arch/riscv/include.

**Comment by maroueneboubakri** [comment]
> @maroueneboubakri @jenswi-linaro riscv_macros.S and riscv.h are updated in #5714 ，and I also cited the changes in this update，Whether wait #5714 merged, then move the two files to the lib/libutils/ext/arch/riscv/include.

Not sure if you read my comments above but this is not needed at all.

**Comment by liushiwei007** [comment]
What is not needed?  this PR  or this move of the header file?

> > @maroueneboubakri @jenswi-linaro riscv_macros.S and riscv.h are updated in #5714 ，and I also cited the changes in this update，Whether wait #5714 merged, then move the two files to the lib/libutils/ext/arch/riscv/include.
> 
> Not sure if you read my comments above but this is not needed at all.

What is not needed,  this PR  or this move of the header file?

**Comment by maroueneboubakri** [comment]
> What is not needed? this PR or this move of the header file?
> 
> > > @maroueneboubakri @jenswi-linaro riscv_macros.S and riscv.h are updated in #5714 ，and I also cited the changes in this update，Whether wait #5714 merged, then move the two files to the lib/libutils/ext/arch/riscv/include.
> > 
> > 
> > Not sure if you read my comments above but this is not needed at all.
> 
> What is not needed, this PR or this move of the header file?

There is no need to move these header since you don't need to store the panic context because it is already saved.

**Pull request closed without merging**